### PR TITLE
Add terminal profile support

### LINKED ISSUE #33: Add to New Terminal menu
Doesn't appear in the New Terminal menu for some reason.

Is there a way to register the REPL as a VSC terminal?

### PR DESCRIPTION
**Pull request by SNDST00M**
Closes #33.

### CODE REVIEW

**Comment by 4086606** [comment]
Awaiting [`@types/vscode`@v1.58.0](https://www.npmjs.com/package/@types/vscode) bump

**Comment by apommel** [comment]
I feel like this PR tries to do too much things together. Maybe it's better to just focus on the terminal profile here, and have a look at other changes in other PR?

**Comment by 4086606** [comment]
Sure, can roll back to the terminal profile commit

I bet you'll have the first extension using this new feature 😄

**Comment by 4086606** [comment]
Done!

**Comment by 4086606** [comment]
Any ETA on merging this?

**Comment by apommel** [comment]
Yeah sorry I have been busy lately. I'll try to have a look this weekend and merge it if everything is fine.

**Comment by 4086606** [comment]
It's no problem, hope you can get to it when you're ready

**Comment by SNDST00M** [comment]
I've noticed this branch is still in my list, any dice? I've been testing it a lot and it works great

**Comment by apommel** [comment]
Hey sorry for the delay. I was able to take a look at it today. It does not work for me, I guess you did not push the latest version: some variables have wrong names I think, I'll review it.

File: src/extension.ts [hunk 1]
```diff
@@ -201,6 +201,23 @@ export function activate(context: vscode.ExtensionContext) {
 	};
 	context.subscriptions.push(vscode.commands.registerCommand("matlab-interactive-terminal.runMatlabSelection", runMatlabSelection));
 
+	const resetPythonPath = () => {
+		pythonPath = getPythonPath();
```
Comment (apommel) [inline src/extension.ts:205]: Do you mean `python_path`?

File: src/extension.ts [hunk 2]
```diff
@@ -201,6 +201,23 @@ export function activate(context: vscode.ExtensionContext) {
...
+	};
+	context.subscriptions.push(vscode.workspace.onDidChangeConfiguration(resetPythonPath));
+
+	const matlabTerminalProfile: vscode.TerminalProfileProvider = {
+		provideTerminalProfile() {
+			return {
+				options: {
+					...terminalLaunchOptions,
+					shellArgs: [terminalPath]
```
Comment (apommel) [inline src/extension.ts:214]: Do you mean `terminal_path`?

Comment (SNDST00M) [reply src/extension.ts:215]: Resolved in 21058a1

File: src/extension.ts (hunk 1 above)
Comment (SNDST00M) [reply src/extension.ts:205]: Yup! Resolved in 21058a1. All variables use camelCase for a more JS/TS and less Pythonic/Lua feel

Comment (apommel) [reply src/extension.ts:205]: Yeah I've never been really consistent with that

**Comment by apommel** [comment]
Okay it works now but I found an issue in the new feature: if I try to open a MATLAB terminal from the integrated terminal without having activated the extension before (so without being on a .m file for example), it fails with the following error `No terminal profile provider registered for id "matlab-interactive-terminal.terminal-profile"`.
I guess it was to be expected. Isn't there a way to activate the extension when the terminal is created through this feature?

**Comment by apommel** [comment]
Actually I just realized that the commit which makes the extension activate on `.m` file is not in this PR so I am not sure why it fails only when no file is opened.

**Comment by SNDST00M** [comment]
We'd make it activate on `*` and limit the commands with `when` clause contexts

File: package.json [hunk 3]
```diff
@@ -18,10 +18,7 @@
 		"Other"
 	],
 	"activationEvents": [
-		"onLanguage:matlab",
-		"onCommand:matlab-interactive-terminal.openMatlabTerminal",
-		"onCommand:matlab-interactive-terminal.runMatlabScript",
-		"onCommand:matlab-interactive-terminal.runMatlabSelection"
+		"*"
```
Comment (apommel) [inline package.json:21]: I don't like that so much, but if there's no choice I guess this would be better:
```suggestion
		"onStartupFinished"
```

**Comment by apommel** [comment]
It's fine with me like this. I'll merge it. However if at some point there is an activation event such as `"onTerminalProfile"` (I saw there was such requests) it would be better to revert the changes to extension activation and add this.
Thanks for the work!

**Comment by SNDST00M** [comment]
https://code.visualstudio.com/updates/v1_57 mentions `onTerminalProfile` as working, let's use it!

**Pull request merged**